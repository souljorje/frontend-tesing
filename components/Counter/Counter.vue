<template>
  <div>
    <form action="#" @submit.prevent="mutate('decrement')">
      <label>
        Decrement by:
        <input type="number" v-model.number="decrementValue">
      </label>
      <button type="submit">Make!</button>
    </form>
    <form action="#" @submit.prevent="mutate('increment')">
      <label>
        Increment by:
        <input type="number" v-model.number="incrementValue">
      </label>
      <button type="submit">Make!</button>
    </form>
    <div>
      Mutate by 1:
      <button @click="decrement(1)">Decrement</button>
      <button @click="increment(1)">Increment</button>
    </div>
    <button @click="reset">Reset</button>
    <div class="value">{{ value }}</div>
  </div>
</template>

<script>
export default {
  name: 'counter',
  data() {
    return {
      value: this.initialValue,
      decrementValue: 0,
      incrementValue: 0,
    };
  },
  props: {
    initialValue: {
      type: Number,
      default: 0,
    },
  },
  watch: {
    value() {
      this.emit('valueChanged', this.value);
    },
  },
  methods: {
    increment(value = 1) {
      this.value += value;
    },
    decrement(value = 1) {
      this.value -= value;
    },
    mutate(actionType) {
      this[actionType](this[`${actionType}Value`]);
      this[`${actionType}Value`] = null;
    },
    reset() {
      this.value = 0;
    },
  },
};
</script>
