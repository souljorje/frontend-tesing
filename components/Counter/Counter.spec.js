import { mount } from '@vue/test-utils';
import Counter from './Counter';

describe('Counter', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Counter);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('renders correctly', () => {
    const wrapper = mount(Counter);
    expect(wrapper.element).toMatchSnapshot();
  });
});
