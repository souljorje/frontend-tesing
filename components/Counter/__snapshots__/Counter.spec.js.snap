// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<div>
  <form
    action="#"
  >
    <label>
      
      Decrement by:
      
      <input
        type="number"
      />
    </label>
     
    <button
      type="submit"
    >
      Make!
    </button>
  </form>
   
  <form
    action="#"
  >
    <label>
      
      Increment by:
      
      <input
        type="number"
      />
    </label>
     
    <button
      type="submit"
    >
      Make!
    </button>
  </form>
   
  <div>
    
    Mutate by 1:
    
    <button>
      Decrement
    </button>
     
    <button>
      Increment
    </button>
  </div>
   
  <button>
    Reset
  </button>
   
  <div>
    0
  </div>
</div>
`;
